app/controllers/articles_controller.rb

```diff
@@ -5,12 +5,71 @@ def index
     # session[:current_search] = params
     @topic_search = params[:topic_search]
 
-    if Topic.pluck(:name).include?(@topic_search)
-      @articles = Topic.find_by_name(@topic_search).articles
-      @selected_articles = Article.random_sort(@articles)
-      @topic = Topic.find_by_name(@topic_search)
+    if Topic.pluck(:name).include?(@topic_search) # topic already exists
+      if Time.now - Topic.find_by_name(@topic_search).articles.last.created_at < 86400 # last article was added to the topic less than 24 hours
```

Comment:
It would be convenient to use an human readable format instead of `86400` (mainly for @Isauret 😁)